### DISCUSSION TITLE
Is Store API not supported yet?

### DISCUSSION THREAD

**Discussion by HanhoKim** [adrien2p/medusa-extender#190] [Q&A]
I am a newbie of medusa and medusa-extender. 

I found out that I can overwrite middleware and some endpoints of Admin API has,
but for the Store APIs, such as **/store/auth**, @Middleware or @Router decoration is not working as intended.

```typescript
@Middleware({ requireAuth: false, routes: [{ method: 'post', path: '/store/auth' }] })
export default class CustomerAuthorizationMiddleware implements MedusaMiddleware {
  public consume(req: Request, res: Response, next: NextFunction): void | Promise<void> {
    console.log('middleware Works?');
    return next();
  }
}
```
The above middleware is not called on [POST] /store/auth

**Comment by HanhoKim** [comment]
But @Middleware having { path: '*' } is working for **[POST] /store/auth**.

```typescript
@Middleware({ requireAuth: false, routes: [{ method: 'post', path: '*' }] })
export default class CustomerAuthorizationMiddleware implements MedusaMiddleware {
  public consume(req: Request, res: Response, next: NextFunction): void | Promise<void> {
    console.log('middleware Works?');
    return next();
  }
}
```

**Reply by HanhoKim** [reply]
I found out that only **[GET] /store/auth**, **[POST] /store/auth**, **[DELETE] /store/auth** is not working. Every other endpoints are working very well.

**Reply by adrien2p** [reply]
Yeah it is normal, they are avoided as they are managed differently in the core. Just to let you know, the extender will be deprecated as we have ported all the features to the medusa 💪

**Reply by HanhoKim** [reply]
@adrien2p So you mean, we can use medusa extender's all features from latest medusa? Great. From which version?
